packages/notification-center/src/hooks/useUnreadCount.ts

```diff
@@ -0,0 +1,87 @@
+import { useEffect } from 'react';
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query';
+import debounce from 'lodash.debounce';
+import { WebSocketEventEnum } from '@novu/shared';
+
+import type { ICountData } from '../shared/interfaces';
+import { useNovuContext } from './useNovuContext';
+import { useSetQueryKey } from './useSetQueryKey';
+import { useFetchNotificationsQueryKey } from './useFetchNotificationsQueryKey';
+import { useUnseenCountQueryKey } from './useUnseenCountQueryKey';
+import { useDataRef } from './useDataRef';
+import { FEED_UNSEEN_COUNT_QUERY_KEY } from './queryKeys';
+import { useUnreadCountQueryKey } from './useUnreadCountQueryKey';
+
+const dispatchUnreadCountEvent = (count: number) => {
+  document.dispatchEvent(new CustomEvent('novu:unread_count_changed', { detail: count }));
+};
+
+/**
+ * TODO: This is a temporary fix
+ * Cypress is slow, so we need to increase the debounce time
+ * This is happening in a very high thruput of updates in testing env.
+ *
+ * Can also happen in real scenarios, so we need to review how we handle concurrency in the future
+ */
+const DEBOUNCE_TIME = typeof window !== 'undefined' && (window as any)?.Cypress ? 1000 : 100;
+
+export const useUnreadCount = ({ onSuccess, ...restOptions }: UseQueryOptions<ICountData, Error, ICountData> = {}) => {
```

Comment:
this hook is basically the copy of the `useUnseen` hook, but with few differences:
- listens to the `unread_count_changed` ws event
- when event comes in it refetches also the unseen count